### PR TITLE
feat: cap contribution based rewards by factored taker fees paid

### LINKED ISSUE #10517: Taker rewards capped by trading fees reward scheme
In order to enhance the reward scheme the following spec change will add the following:

When there is an amount to be distributed as taker rewards:

- there are keys that have paid the relevant fees; we calculate the fraction of 1 of relevant fees each eligible key paid/received.
- each key then gets max(fraction times total, taker fees paid in epoch times conversion factor from quantum to reward asset).
- whatever is left over stays in the reward pool for next epoch (if this is valid) or sent to the key that funded the pool or whatever makes most sense in the current design.

### Spec

- https://github.com/vegaprotocol/specs/issues/2142

### PR DESCRIPTION
**Pull request by ze97286**
closes #10517

### CODE REVIEW

**Comment by jeremyletang** [comment]
Let's no merge this before the final release for 74 is cut.

File: protos/sources/vega/vega.proto [hunk 1]
```diff
@@ -1126,6 +1126,8 @@ message DispatchStrategy {
   DistributionStrategy distribution_strategy = 12;
   // Ordered list, using start rank, defining the rank bands and share ratio for each band. Mandatory for the rank distribution strategy.
   repeated Rank rank_table = 13;
+  // if set, the actual amount of reward transferred to each public key during distribution for this transfer will be `min(calculated_reward_in_quantum, cap_reward_fee_multiple × fees_paid_this_epoch_in_quantum).
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1129]:
```suggestion
  // If set, the actual amount of rewards transferred to each public key during distribution for this transfer will be `min(calculated_reward_in_quantum, cap_reward_fee_multiple × fees_paid_this_epoch_in_quantum).
```

File: protos/sources/vega/vega.proto [hunk 2]
```diff
@@ -1126,6 +1126,8 @@ message DispatchStrategy {
...
+  optional string cap_reward_fee_multiple = 14;
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1130]: Please add this to the GraphQL schema too

File: datanode/gateway/graphql/schema.graphql [hunk 3]
```diff
@@ -1446,6 +1446,9 @@ type DispatchStrategy {
 
   "Ascending order list of start rank and corresponding share ratio"
   rankTable: [RankTable]
+
+  "Optional multiplier on taker fees used to cap the reward a party may receive in an epoch"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:1450]:
```suggestion
  "Optional multiplier on taker fees used to cap the rewards a party may receive in an epoch"
```

**Review by candida-d — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**